helita/sim/ebysus.py

```diff
@@ -150,108 +302,590 @@ def _get_simple_var(self, var, mf_ispecies=0, mf_ilevel=0, order='F', mode='r',
+                    mode='r', *args, **kwargs):
+
+        self.iix = iix
+        self.iiy = iiy
+        self.iiz = iiz
+
+        try:
+            if ((snap is not None) and  any(snap != self.snap)):
+                self.set_snap(snap)
+
+        except ValueError:
+            if ((snap is not None) and any(snap != self.snap)):
+                self.set_snap(snap)
+
+        if var in self.varsmfc:
+            if mf_ilevel is None and self.mf_ilevel == 1:
+                mf_ilevel = 2
+                print("Warning: mfc is only for ionized species,"
+                      "Level changed to 2")
+            if mf_ilevel == 1:
+                mf_ilevel = 2
+                print("Warning: mfc is only for ionized species."
+                      "Level changed to 2")
+
+        if var not in self.snapevars:
+            if (mf_ispecies is None):
+                if self.mf_ispecies < 1:
+                    mf_ispecies = 1
+                    print("Warning: variable is only for electrons,"
+                          "iSpecie changed to 1")
+            elif (mf_ispecies < 1):
+                mf_ispecies = 1
+                print("Warning: variable is only for electrons,"
+                      "iSpecie changed to 1")
+
+        if (((mf_ispecies is not None) and (
+                mf_ispecies != self.mf_ispecies)) or ((
+                mf_ilevel is not None) and (mf_ilevel != self.mf_ilevel))):
+            self.set_mfi(mf_ispecies, mf_ilevel)
+
+        # lengths for dimensions of return array
+        self.xLength = 0
+        self.yLength = 0
+        self.zLength = 0
+
+        for dim in ('iix', 'iiy', 'iiz'):
+            if getattr(self, dim) is None:
+                if dim[2] == 'z':
+                    setattr(self, dim[2] + 'Length', getattr(self, 'n' + dim[2]+'b'))
+                else:
+                    setattr(self, dim[2] + 'Length', getattr(self, 'n' + dim[2]))
+                setattr(self, dim, slice(None))
+            else:
+                indSize = np.size(getattr(self, dim))
+                setattr(self, dim[2] + 'Length', indSize)
+        print(self.zLength,self.nzb)
+        snapLen = np.size(self.snap)
+        value = np.empty([self.xLength, self.yLength, self.zLength, snapLen])
+
+        for i in range(0, snapLen):
+            self.snapInd = 0
+            self._set_snapvars()
+            self._init_vars()
+            value[:, :, :, i] = self.get_var(
+                var, snap=snap[i], iix=self.iix, iiy=self.iiy, iiz=self.iiz,
+                mf_ispecies = self.mf_ispecies, mf_ilevel=self.mf_ilevel)
+
+        try:
+            if ((snap is not None) and (snap != self.snap)):
+                self.set_snap(snap)
+
+        except ValueError:
+            if ((snap is not None) and any(snap != self.snap)):
+                self.set_snap(snap)
+
+        return value
+
+
+###########
+#  TOOLS  #
+###########
+
+
+def read_mftab_ascii(filename):
+    '''
```

Comment:
Again, is this different from `read_idl_ascii`? Why not just modify the output of `read_idl_ascii`?